src/chainparams.cpp

```diff
@@ -466,14 +474,57 @@ class CRegTestParams : public CChainParams {
     /**
      * Allows modifying the Version Bits regtest parameters.
      */
-    void UpdateVersionBitsParameters(Consensus::DeploymentPos d, int64_t nStartTime, int64_t nTimeout)
+    void UpdateVersionBitsParameters(Consensus::DeploymentPos d, int64_t start_height, int64_t timeout_height, int64_t min_activation_height)
     {
-        consensus.vDeployments[d].nStartTime = nStartTime;
-        consensus.vDeployments[d].nTimeout = nTimeout;
+        consensus.vDeployments[d].startheight = start_height;
+        consensus.vDeployments[d].timeoutheight = timeout_height;
+        consensus.vDeployments[d].m_min_activation_height = min_activation_height;
     }
+    bool CheckVBitsHeights(std::string& error, int64_t startheight, int64_t timeoutheight, int64_t min_activation_height);
     void UpdateActivationParametersFromArgs(const ArgsManager& args);
 };
 
+bool CRegTestParams::CheckVBitsHeights(std::string& error, int64_t startheight, int64_t timeoutheight, int64_t min_activation_height)
+{
+    // Special always or never active cases
+    if ((startheight == Consensus::BIP9Deployment::NEVER_ACTIVE && timeoutheight == Consensus::BIP9Deployment::NEVER_ACTIVE)
+        || (startheight == Consensus::BIP9Deployment::ALWAYS_ACTIVE && timeoutheight == Consensus::BIP9Deployment::NO_TIMEOUT)) {
+        return true;
+    }
+    if ((startheight == Consensus::BIP9Deployment::NEVER_ACTIVE && timeoutheight != Consensus::BIP9Deployment::NEVER_ACTIVE)
+        || (startheight != Consensus::BIP9Deployment::NEVER_ACTIVE && timeoutheight == Consensus::BIP9Deployment::NEVER_ACTIVE)) {
+        error = strprintf("When one of startheight or timeoutheight is %d, both must be %d", Consensus::BIP9Deployment::NEVER_ACTIVE, Consensus::BIP9Deployment::NEVER_ACTIVE);
+        return false;
+    }
+    if (startheight == Consensus::BIP9Deployment::ALWAYS_ACTIVE && timeoutheight != Consensus::BIP9Deployment::NO_TIMEOUT) {
+        error = strprintf("When startheight is %d, timeoutheight must be %d", Consensus::BIP9Deployment::ALWAYS_ACTIVE, Consensus::BIP9Deployment::NO_TIMEOUT);
```

Comment:
This isn't true, and specifying NO_TIMEOUT is kinda annoying...?

Reply:
Removed it.